src/uu/factor/src/factor.rs

```diff
@@ -70,16 +70,22 @@ impl PartialEq for Decomposition {
 impl Eq for Decomposition {}
 
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub struct Factors(RefCell<Decomposition>);
+pub struct Factors {
+    decomposition: RefCell<Decomposition>,
+    print_exponents: bool,
```

Comment:
hmm, is carrying the boolean all around really necessary ?

Reply:
@sylvestre what do you think?

Reply:
Much better indeed

Reply:
@anastygnome clever approach, I learned something new :)

Reply:
@cakebaker Yeah, I stumbled upon that while looking for something the other day. 
As a sidenote, please comment such clever tricks if you ever use it in code 
Glad you learned something new, and happy Easter!

Reply:
@cakebaker Thanks for the feedback and the information about the alternate format in Rust's Formatter. I appreciate your suggestion, and I will  use the alternate format instead of adding a boolean to the factor struct.

Also, recarding the use of `repeat()` inside the `write!()` macro, it cannot be directly used because `write!()` expects its arguments to be expressions that can be evaluated at compile-time, whereas `repeat()` is a method that requires runtime evaluation. This is because `repeat()` involves creating a new `String` with repeated copies of the original string, which cannot be done at compile-time.

So I would need to do it like this:

```Rust
impl fmt::Display for Factors {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        for (p, exp) in v.iter() {
            let include_exponents = f.alternate();
            if include_exponents && *exp  > 1 {
                write!(f, " {p}^{exp}",)?;
            } else {
                let output: &String = &format!(" {}",p).repeat(*exp as usize);
                write!(f, "{output}")?;
            }
        }
    }
}
```

And this could result in higher memory usage and slower performance, especially when `exp` is large due to memory allocation and string manipulation.

Reply:
Hey, thanks for looking into the repeat version! 


I was indeed mistaken to use repeat, yet we can avoid allocations, using the following idiom : 

```rust
else{
     write!(f, "{}", std::iter::repeat(format!(" {}", p)).take(*exp as usize).collect::<String>())?;
 }
```
Which is what I had in mind.

Reply:
I ran benchmark tests using the command `seq 2 $((10 ** 8)) | factor`. The average elapsed time for executing this 10 times with the `alternate()` function and `repeat()` was 108422 ms, which is 30% slower compared to using the `for` loop with the `alternate()` function, which had an average elapsed time of 83422 ms. 
So, in this situation, it appears that continuing with the `for` loop is better as it seems to be faster.

Reply:
@Ideflop, I don't know how you benched it, but a simple timing is a trap on those apps, there is great variability across runs and both methods are the same, really
```shell
hyperfine --shell=zsh "for i in {1..10000} ; do ./target/release/factor_fmt  $i; done" "for i in {1..10000} ; do ./target/release/factor_for  $i; done"
Benchmark 1: for i in {1..10000} ; do ./target/release/factor_fmt  ; done
  Time (mean ± σ):     16.859 s ±  8.278 s    [User: 8.346 s, System: 8.771 s]
  Range (min … max):   10.162 s … 30.896 s    10 runs
 
Benchmark 2: for i in {1..10000} ; do ./target/release/factor_for  ; done
  Time (mean ± σ):     15.710 s ±  1.880 s    [User: 8.575 s, System: 7.346 s]
  Range (min … max):   12.902 s … 17.785 s    10 runs
 
Summary
  'for i in {1..10000} ; do ./target/release/factor_for  ; done' ran
    1.07 ± 0.54 times faster than 'for i in {1..10000} ; do ./target/release/factor_fmt  ; done'
```

So this is a matter of taste, it seems :)

Reply:
Indeed, there was a problem. I obtained the following results using hyperfine:
```bash
hyperfine --shell=bash "for i in {1..10000} ; do ./coreutils-old factor $i; done" "for i in {1..10000} ; do ./coreutils-for factor $i; done" "for i in {1..10000} ; do ./coreutils-repeat factor $i; done"
Benchmark 1: for i in {1..10000} ; do ./coreutils-old factor ; done
  Time (mean ± σ):     19.188 s ±  1.018 s    [User: 10.589 s, System: 8.938 s]
  Range (min … max):   18.167 s … 21.210 s    10 runs

Benchmark 2: for i in {1..10000} ; do ./coreutils-for factor ; done
  Time (mean ± σ):     18.754 s ±  1.206 s    [User: 10.459 s, System: 8.651 s]
  Range (min … max):   17.593 s … 21.410 s    10 runs

Benchmark 3: for i in {1..10000} ; do ./coreutils-repeat factor ; done
  Time (mean ± σ):     19.741 s ±  1.005 s    [User: 10.860 s, System: 9.217 s]
  Range (min … max):   18.215 s … 21.357 s    10 runs

Summary
  'for i in {1..10000} ; do ./coreutils-for factor ; done' ran
    1.02 ± 0.09 times faster than 'for i in {1..10000} ; do ./coreutils-old factor ; done'
    1.05 ± 0.09 times faster than 'for i in {1..10000} ; do ./coreutils-repeat factor ; done'
```

Using `repeat` is 5% slower (not 30%) compared to the `for` loop.  Furthermore, the `for` loop was faster than the current implementation of `uutils factor`. At the moment, I will stick with the `for` loop, therefore.

Reply:
Yes, I have tested it with the release branch. After conducting further tests, I found no difference between the current implementation of `factor` and the new one. However, I observed that using `repeat` was always slightly slower. Therefore, I have decided to keep the for loop instead of replacing it with `repeat` because the current implementation of `factor` from uutils is slower than the GNU version.